Cabal-QuickCheck/src/Test/QuickCheck/Instances/Cabal.hs

```diff
@@ -503,14 +503,25 @@ shortListOf1 bound gen = sized $ \n -> do
     vectorOf k gen
 
 arbitraryShortToken :: Gen String
-arbitraryShortToken = arbitraryShortStringWithout "{}[]"
+arbitraryShortToken = arbitraryShortStringWithout "{}[]" (const True)
 
 arbitraryShortPath :: Gen String
-arbitraryShortPath = arbitraryShortStringWithout "{}[],"
-
-arbitraryShortStringWithout :: String -> Gen String
-arbitraryShortStringWithout excludeChars =
-    shortListOf1 5 $ elements [c | c <- ['#' ..  '~' ], c `notElem` excludeChars ]
+arbitraryShortPath = arbitraryShortStringWithout "{}[],<>:|*?" (not . winDevice)
+    where
+        devices = ["con", "aux", "prn", "com", "lpt", "nul"]
+        winDevice :: String -> Bool
+        winDevice p = let p' = map toLower p in any (isDev p') devices
+        isDev :: String -> String -> Bool
+        isDev p s = s `isPrefixOf` p || ('\\' : s) `isInfixOf` p
```

Comment:
I wish I knew what `s` and `p` stand for 🥲

Reply:
`string` and `path`. Likewise `p` and `p'` are `path`s. I can expand those out, but I tend to go short for utility helpers like those.